src/crypto/tls/handshake_server_test.go

```diff
@@ -539,6 +539,41 @@ func testCrossVersionResume(t *testing.T, version uint16) {
 	}
 }
 
+func TestTLSFlags(t *testing.T) {
+	serverConfig := &Config{
+		Certificates: []Certificate{{
+			Certificate: [][]byte{testRSACertificate},
+			PrivateKey:  testRSAPrivateKey,
+		}},
+		TLSFlagsSupported: []TLSFlag{0x50},
+	}
+	clientCert, err := X509KeyPair([]byte(clientECDSACertificatePEM), []byte(clientECDSAKeyPEM))
+	if err != nil {
+		t.Fatalf("couldn't load client certs")
+	}
+	clientConfig := &Config{
+		TLSFlagsSupported:  []TLSFlag{0x50},
```

Comment:
Suggestion: Generate a flags list with multiple flags in it. Also, it might be good to choose flags at random so that we excercise the encoding more comprehensively.